tests/test_asg.py

```diff
@@ -329,3 +331,48 @@ def test_asg_security_group(self):
         resources = p.run()
         self.assertEqual(len(resources), 1)
         self.assertEqual(resources[0]['AutoScalingGroupName'], 'ContainersFTW')
+
+    def test_asg_activity_filter(self):
+        session = self.replay_flight_data('test_asg_launch_activity_filter')
+        self.patch(LaunchActivityFilter, 'executor_factory', MainThreadExecutor)
+        p = self.load_policy({
+            'name': 'asg-activities',
+            'resource': 'asg',
+            'filters': [{
+                'or': [
+                    {'type': 'launch-activity', 'status': 'Failed'},
+                    {'type': 'launch-activity', 'status': 'Successful'}]
```

Comment:
this implementation will effectively pull the full set of activities twice per resource when given this policy.